src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsrestrictedpatientsapi/services/BatchReleaseDateRemoval.kt

```diff
@@ -0,0 +1,36 @@
+package uk.gov.justice.digital.hmpps.hmppsrestrictedpatientsapi.services
+
+import com.microsoft.applicationinsights.TelemetryClient
+import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.hmppsrestrictedpatientsapi.gateways.PrisonerSearchApiGateway
+import uk.gov.justice.digital.hmpps.hmppsrestrictedpatientsapi.repositories.RestrictedPatientsRepository
+import java.time.Clock
+import java.time.LocalDate
+
+@Service
+class BatchReleaseDateRemoval(
+  private val restrictedPatientsRepository: RestrictedPatientsRepository,
+  private val prisonerSearchApiGateway: PrisonerSearchApiGateway,
+  private val telemetryClient: TelemetryClient,
+  private val clock: Clock,
+) {
+
+  fun removeNonLifePrisonersPastConditionalReleaseDate() {
+    val now = LocalDate.now(clock)
+    val toBeDeleted = restrictedPatientsRepository.findAll()
+      .chunked(1000)
+      .flatMap { chunk ->
+        val results = prisonerSearchApiGateway.findByPrisonNumbers(chunk.map { it.prisonerNumber })
+        results.filter { it.indeterminateSentence == false && it.conditionalReleaseDate?.isBefore(now) ?: false }
+          .map { it.prisonerNumber }
+      }
+
+    if (toBeDeleted.isNotEmpty()) {
+      telemetryClient.trackEvent(
```

Comment:
at present just create an event for the removal, rather than actually removing any patients